### PR TITLE
Edited user model, added registration and log in forms

### PR DESCRIPTION
**Pull request by mayasa3**
- Renamed login app as accounts to prevent naming issues and updated dependencies

- Renamed variables

- Updated UserAccount model to extend base User model

- Updated models in searchJob and user_notification apps to fix an issue with foreign keys referencing users

- Forms, views and URLs for registration, log in and log out features

- HTML pages for registration and login

@bardielle @benipeled @guy9050

### CODE REVIEW

**Review by bardielle — CHANGES_REQUESTED** [review]
Split it into multiple commits
it cannot be read like that...

**Comment by mayasa3** [comment]
I split it into 2 commits, is it ok like this? @bardielle

File: accounts/admin.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+UserAdmin.add_form = SignUpForm
+UserAdmin.add_fieldsets = (
+    (None, {
+        'classes': ('wide',),
+        'fields': ('username', 'email', 'password1', 'password2', 'first_name', 'last_name',)
+    }),
+)
+
+admin.site.unregister(User)
+admin.site.register(User, UserAdmin)
```
Comment (bardielle) [inline accounts/admin.py]: @ifireball is it fine to unregister user like that? Why do we need it?

File: searchJob/models.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
+from django.db import models
+
+
+class Company_type(models.TextChoices):
```
Comment (bardielle) [inline searchJob/models.py]: class names are with capital letters... Company_type -> CompanyType

File: searchJob/models.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    BIOTECH = 'BIOTECH', 'BIOTECH'
+    MEDTECH = 'MEDTECH', 'MEDTECH'
+    FOODTECH = 'FOODTECH', 'FOODTECH'
+    AUTOMATIVE = 'AUTOMATIVE', 'AUTOMATIVE'
+    SW = 'SW', 'SW'
+    HW = 'HW', 'HW'
+    UNSPECIFIED = 'UNSPECIFIED', 'UNSPECIFIED'
+
+
+class Job_scope(models.TextChoices):
```
Comment (bardielle) [inline searchJob/models.py]: Job_scope ->JobScope

File: searchJob/models.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    FULL = 'FULL', 'FULL'
+    PART = 'PART', 'PART'
+    INTER = 'INTER', 'INTER'
+    STUDENT = 'STUDENT', 'STUDENT'
+    UNSPECIFIED = 'UNSPECIFIED', 'UNSPECIFIED'
+
+
+class Application_status(models.TextChoices):
```
Comment (bardielle) [inline searchJob/models.py]: Application_status ->ApplicationStatus

File: searchJob/models.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    job_scope = models.CharField(max_length=50, choices=Job_scope.choices, default=Job_scope.UNSPECIFIED,
+                                 blank=True, null=True)
+    job_URL = models.URLField(max_length=200)
+
+    def __str__(self):
+        return self.job_name
+
+
+class UserApplication(models.Model):
+    UserAccount_id = models.ForeignKey("accounts.UserAccount", on_delete=models.CASCADE, blank=True, null=True)
```
Comment (bardielle) [inline searchJob/models.py]: UserAccount_id ->user_account_id

File: searchJob/models.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    jobPost_id = models.ForeignKey(JobPost, on_delete=models.CASCADE, blank=True, null=True)
```
Comment (bardielle) [inline searchJob/models.py]: job_post_id

File: accounts/urls.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
+"""joboard URL Configuration
+
+The `urlpatterns` list routes URLs to views. For more information please see:
+    https://docs.djangoproject.com/en/3.1/topics/http/urls/
+Examples:
```
Comment (bardielle) [inline accounts/urls.py]: remove un-relevant comments....

File: accounts/views.py [hunk 8]
```diff
@@ -1,3 +1,44 @@
...
\ No newline at end of file
+
+def signup_view(request):
+    if request.method == 'POST':
+        user_form = SignUpForm(request.POST)
+        account_form = AccountForm(request.POST)
+        if user_form.is_valid() and account_form.is_valid():
+            user = user_form.save()
+            user.save()
+            useraccount = account_form.save(commit=False)
```
Comment (bardielle) [inline accounts/views.py]: useraccount - >user_account

**Review by bardielle — CHANGES_REQUESTED** [review]

**Comment by mayasa3** [comment]
Renamed everything and changed the admin file

File: accounts/views.py (hunk 8 above)
Comment (bardielle) [reply accounts/views.py]: @mayasa3

File: accounts/admin.py [hunk 9]
```diff
@@ -34,5 +14,3 @@ class UserAdmin(BaseUserAdmin):
     }),
 )
 
-admin.site.unregister(User)
```
Comment (bardielle) [inline accounts/admin.py]: fix my comments on the commit it was mentioned...

File: searchJob/models.py [hunk 10]
```diff
@@ -1,7 +1,7 @@
 from django.db import models
 
 
-class Company_type(models.TextChoices):
+class CompanyType(models.TextChoices):
```
Comment (bardielle) [inline searchJob/models.py:5]: fix my comments on the commit it was mentioned...

File: searchJob/models.py [hunk 11]
```diff
@@ -12,15 +12,15 @@ class Company_type(models.TextChoices):
     UNSPECIFIED = 'UNSPECIFIED', 'UNSPECIFIED'
 
 
-class Job_scope(models.TextChoices):
+class JobScope(models.TextChoices):
```
Comment (bardielle) [inline searchJob/models.py:16]: fix my comments on the commit it was mentioned...

File: searchJob/models.py [hunk 12]
```diff
@@ -12,15 +12,15 @@ class Company_type(models.TextChoices):
...
     FULL = 'FULL', 'FULL'
     PART = 'PART', 'PART'
     INTER = 'INTER', 'INTER'
     STUDENT = 'STUDENT', 'STUDENT'
     UNSPECIFIED = 'UNSPECIFIED', 'UNSPECIFIED'
 
 
-class Application_status(models.TextChoices):
+class ApplicationStatus(models.TextChoices):
```
Comment (bardielle) [inline searchJob/models.py:24]: fix my comments on the commit it was mentioned...

File: searchJob/models.py [hunk 13]
```diff
@@ -55,8 +55,8 @@ def __str__(self):
 
 
 class UserApplication(models.Model):
-    UserAccount_id = models.ForeignKey("accounts.UserAccount", on_delete=models.CASCADE, blank=True, null=True)
-    jobPost_id = models.ForeignKey(JobPost, on_delete=models.CASCADE, blank=True, null=True)
+    user_account_id = models.ForeignKey("accounts.UserAccount", on_delete=models.CASCADE, blank=True, null=True)
```
Comment (bardielle) [inline searchJob/models.py]: fix my comments on the commit it was mentioned...

**Review by bardielle — CHANGES_REQUESTED** [review]

File: accounts/admin.py (hunk 1 above)
Comment (mayasa3) [reply accounts/admin.py]: A lot of the instructions I found for adding fields to the admin panel said to unregister and re-register it, but I can remove that part

File: accounts/admin.py (hunk 9 above)
Comment (mayasa3) [reply accounts/admin.py]: Does it look ok now? Had some trouble with the rebase

**Comment by mayasa3** [comment]
Updated the commits, is there anything else left to fix?

File: accounts/admin.py (hunk 1 above)
Comment (bardielle) [reply accounts/admin.py]: Please fix it

File: searchJob/models.py (hunk 6 above)
Comment (bardielle) [reply searchJob/models.py]: ??

File: searchJob/models.py (hunk 5 above)
Comment (bardielle) [reply searchJob/models.py]: ??

File: accounts/urls.py (hunk 7 above)
Comment (bardielle) [reply accounts/urls.py]: ??

File: accounts/views.py (hunk 8 above)
Comment (bardielle) [reply accounts/views.py]: @mayasa3 this comment is still relevant

**Review by bardielle — CHANGES_REQUESTED** [review]

File: accounts/admin.py [hunk 14]
```diff
@@ -0,0 +1,35 @@
+from django.contrib import admin
+from django.contrib.auth.admin import UserAdmin as BaseUserAdmin
+from django.contrib.auth.models import User
+from accounts.models import UserAccount
+
+
+admin.site.register(UserAccount)
```
Comment (bardielle) [inline accounts/admin.py:5]: Move it to the end of the file where all the other registers

File: accounts/admin.py [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+
+
+UserAdmin.add_fieldsets = (
+    (None, {
+        'classes': ('wide',),
+        'fields': ('username', 'email', 'password1', 'password2', 'first_name', 'last_name',)
+    }),
+)
+
+admin.site.unregister(User)
```
Comment (bardielle) [inline accounts/admin.py]: Why do you need "unregister" and again "register"?

File: accounts/admin.py [hunk 16]
```diff
@@ -0,0 +1,35 @@
+from django.contrib import admin
+from django.contrib.auth.admin import UserAdmin as BaseUserAdmin
+from django.contrib.auth.models import User
+from accounts.models import UserAccount
+
+
+admin.site.register(UserAccount)
+
+
+class AccountInline(admin.StackedInline):
```
Comment (bardielle) [inline accounts/admin.py]: Why all those classes in the admin.py file? they need to be in the models.py
@ifireball Am I wrong?

File: accounts/migrations/0003_auto_20201213_1916.py [hunk 17]
```diff
@@ -0,0 +1,55 @@
+# Generated by Django 3.1.4 on 2020-12-13 19:16
+
```
Comment (bardielle) [inline accounts/migrations/0003_auto_20201213_1916.py:2]: You dont need multiple migration files as far as I know
@ifireball fix me if Im wrong....

**Pull request closed without merging**